File: 2085-array-with-elements-not-equal-to-average-of-neighbors/2085-array-with-elements-not-equal-to-average-of-neighbors.py
class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        nums.sort()

        for i in range(0, len(nums) // 2, 2):
            key = (len(nums) - 1 - i if len(nums) % 2 == 0 else len(nums) - 2 - i)
            nums[i], nums[key] = nums[key], nums[i]

        return nums

File: 2085-array-with-elements-not-equal-to-average-of-neighbors/2085-array-with-elements-not-equal-to-average-of-neighbors.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        
        vector<int> ans;

        int start = 0;
        int end = nums.size() - 1;

        while (start < end) {
            ans.push_back(nums[start++]);
            ans.push_back(nums[end--]);
        }

        if (start == end) {
            ans.push_back(nums[start]);
        }

        return ans;
    }
};